src/phoenix/db/migrations/future_versions/cd164e83824f_users_and_tokens.py

```diff
@@ -0,0 +1,299 @@
+    role: Mapped[str] = mapped_column(
+        CheckConstraint("role IN ('SYSTEM', 'ADMIN', 'GENERAL')", name="valid_role"),
+        unique=True,
+    )
+
+
+class User(Base):
+    __tablename__ = "users"
+    id: Mapped[int] = mapped_column(primary_key=True)
+    user_role_id: Mapped[int] = mapped_column(
+        ForeignKey("user_roles.id"),
+        index=True,
+    )
+    username: Mapped[Optional[str]] = mapped_column(nullable=True, unique=True, index=True)
+    email_address: Mapped[Optional[str]] = mapped_column(nullable=True, unique=True, index=True)
+    auth_method: Mapped[str] = mapped_column(
+        CheckConstraint("auth_method IN ('LOCAL')", name="valid_auth_method")
+    )
+    password_hash: Mapped[Optional[str]]
+    reset_password: Mapped[bool]
+    created_at: Mapped[datetime] = mapped_column(UtcTimeStamp, server_default=func.now())
+    updated_at: Mapped[datetime] = mapped_column(
+        UtcTimeStamp, server_default=func.now(), onupdate=func.now()
+    )
+    deleted_at: Mapped[Optional[datetime]] = mapped_column(UtcTimeStamp)
+
+
+# revision identifiers, used by Alembic.
+revision: str = "cd164e83824f"
+down_revision: Union[str, None] = "3be8647b87d8"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+async def insert_roles_and_users(connection: AsyncConnection) -> None:
+    """
+    Populates the `user_roles` table and adds a system user and initial admin
+    user to the `users` table.
+    """
+    await connection.execute(
+        insert(UserRole).values([{"role": "SYSTEM"}, {"role": "ADMIN"}, {"role": "GENERAL"}])
+    )
+    system_user_role_id = sa.select(UserRole.id).where(UserRole.role == "SYSTEM").scalar_subquery()
+    admin_user_role_id = sa.select(UserRole.id).where(UserRole.role == "ADMIN").scalar_subquery()
+    await connection.execute(
+        insert(User).values(
+            [
+                {
+                    "user_role_id": system_user_role_id,
+                    "username": None,
+                    "email_address": None,
+                    "auth_method": "LOCAL",
+                    "password_hash": None,
+                    "reset_password": False,
+                },
+                {
+                    "user_role_id": admin_user_role_id,
+                    "username": "admin",
+                    "email_address": None,
+                    "auth_method": "LOCAL",
+                    "password_hash": None,  # todo: replace this with the hashed PHOENIX_SECRET
+                    "reset_password": True,
+                },
+            ]
+        )
+    )
+
+
+def upgrade() -> None:
+    op.create_table(
+        "user_roles",
+        sa.Column("id", sa.Integer, primary_key=True),
+        sa.Column(
+            "role",
+            sa.String,
+            sa.CheckConstraint(
+                "role IN ('SYSTEM', 'ADMIN', 'GENERAL')",
+                name="valid_role",
```

Comment:
I think in general having a role table probably is more future proof. We don't have to do it now but string based constraints don't scale the greatest.

Reply:
removed check constraint